### PR TITLE
En_Pm

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Pm/z_en_pm.c [hunk 1]
```diff
@@ -1,4 +1,11 @@
+/*
+ * File: z_en_pm.c
+ * Overlay: ovl_En_Pm
+ * Description:
```
Comment (tom-overton) [inline src/overlays/actors/ovl_En_Pm/z_en_pm.c:4]: Looking at it in kz, it seems like this actor is the Postman:

NKZJ01_2021-09-17_16-08-27

```suggestion
 * Description: Postman
```

File: include/functions.h [hunk 2]
```diff
@@ -2680,11 +2680,11 @@ void func_8013A46C(s32 flag);
 u32 func_8013A4C4(s32 flag);
 s16 func_8013A504(s16 val);
 s32 func_8013A530(GlobalContext* globalCtx, Actor* actor, s32 flag, Vec3f* pos, Vec3s* rot, f32 distanceMin, f32 distanceMax, s16 angleError);
-// void func_8013A7C0(void);
+Actor* func_8013A7C0(GlobalContext* globalCtx, s32 arg0);
```
Comment (engineer124) [inline include/functions.h:2683]:
```suggestion
Actor* func_8013A7C0(GlobalContext* globalCtx, s32 arg1);
```

File: src/overlays/actors/ovl_En_Pm/z_en_pm.c [hunk 3]
```diff
@@ -25,128 +206,1705 @@ const ActorInit En_Pm_InitVars = {
...
+
+s32 func_80AF9008(EnPm* this, GlobalContext* globalCtx, struct_80133038_arg2* arg2) {
+    u16 sp56 = gSaveContext.time - 0x3FFC;
+    u8 sp55 = this->actor.params & 0xFF;
+    Actor* sp50;
+    Vec3s* sp4C;
+    Vec3f sp40;
+    Vec3f sp34;
+    s16 temp;
+    s32 ret = 0;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Pm/z_en_pm.c:950]:
```suggestion
    s32 ret = false;
```

File: src/overlays/actors/ovl_En_Pm/z_en_pm.c [hunk 4]
```diff
@@ -25,128 +206,1705 @@ const ActorInit En_Pm_InitVars = {
...
+            this->unk_36E = sp56 - arg2->unk4;
+            this->actor.flags &= ~1;
+            if (gSaveContext.weekEventReg[90] & 8) {
+                this->unk_356 |= 0x800;
+            }
+            this->unk_356 |= 0x9000;
+            this->unk_356 |= 0x200;
+            func_80AF7E98(this, 0);
+            this->actor.gravity = 0.0f;
+            ret = 1;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Pm/z_en_pm.c:984]:
```suggestion
            ret = true;
```

File: src/overlays/actors/ovl_En_Pm/z_en_pm.c [hunk 5]
```diff
@@ -25,128 +206,1705 @@ const ActorInit En_Pm_InitVars = {
...
+    }
+    return ret;
+}
+
+s32 func_80AF91E8(EnPm* this, GlobalContext* globalCtx, struct_80133038_arg2* arg2) {
+    u16 sp2E = (u16)(gSaveContext.time - 0x3FFC);
+    u16 phi_v1;
+    u8 sp2B = this->actor.params & 0xFF;
+    s32 pad;
+    s32 ret = 0;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Pm/z_en_pm.c:995]:
```suggestion
    s32 ret = false;
```

File: src/overlays/actors/ovl_En_Pm/z_en_pm.c [hunk 6]
```diff
@@ -25,128 +206,1705 @@ const ActorInit En_Pm_InitVars = {
...
+    return ret;
+}
+
+s32 func_80AF94AC(EnPm* this, GlobalContext* globalCtx, struct_80133038_arg2* arg2) {
+    u8 sp4F = this->actor.params & 0xFF;
+    Vec3f sp40;
+    Vec3f sp34;
+    Vec3s* sp30;
+    s32 pad;
+    s32 ret = 0;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Pm/z_en_pm.c:1076]:
```suggestion
    s32 ret = false;
```

File: src/overlays/actors/ovl_En_Pm/z_en_pm.c [hunk 7]
```diff
@@ -25,128 +206,1705 @@ const ActorInit En_Pm_InitVars = {
...
+        Math_Vec3f_Copy(&this->actor.world.pos, &sp40);
+        Math_Vec3f_Copy(&this->actor.prevPos, &sp40);
+        if (arg2->unk0 == 24) {
+            Actor_UnsetSwitchFlag(globalCtx, 0);
+            Actor_UnsetSwitchFlag(globalCtx, 1);
+            this->unk_394 = 0;
+            this->unk_368 = 60.0f;
+            func_80AF7E98(this, 9);
+        }
+        ret = 1;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Pm/z_en_pm.c:1098]:
```suggestion
        ret = true;
```

File: src/overlays/actors/ovl_En_Pm/z_en_pm.c [hunk 8]
```diff
@@ -25,128 +206,1705 @@ const ActorInit En_Pm_InitVars = {
...
+    return ret;
+}
+
+s32 func_80AF95E8(EnPm* this, GlobalContext* globalCtx, struct_80133038_arg2* arg2) {
+    u8 sp4F = this->actor.params & 0xFF;
+    Vec3f sp40;
+    Vec3f sp34;
+    Vec3s* sp30;
+    s32 pad;
+    s32 ret = 0;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Pm/z_en_pm.c:1109]:
```suggestion
    s32 ret = false;
```

File: src/overlays/actors/ovl_En_Pm/z_en_pm.c [hunk 9]
```diff
@@ -25,128 +206,1705 @@ const ActorInit En_Pm_InitVars = {
...
+            default:
+                if (arg2->unk0 == 0x1D) {
+                    this->actor.world.rot.y = BINANG_ROT180(this->actor.world.rot.y);
+                }
+                func_8013AED4(&this->unk_356, 3, 7);
+                this->unk_356 |= 0x9000;
+                func_80AF7E98(this, 3);
+                break;
+        }
+        ret = 1;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Pm/z_en_pm.c:1185]:
```suggestion
        ret = true;
```

File: src/overlays/actors/ovl_En_Pm/z_en_pm.c [hunk 10]
```diff
@@ -25,128 +206,1705 @@ const ActorInit En_Pm_InitVars = {
...
+        func_8013AED4(&this->unk_356, 3, 7);
+        this->unk_356 |= 0x20;
+        this->unk_356 |= 0x9000;
+        if (this->unk_258 != 0) {
+            this->unk_356 |= 0x800;
+            func_80AF7E98(this, 5);
+        } else {
+            func_80AF7E98(this, 3);
+        }
+        ret = 1;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Pm/z_en_pm.c:1236]:
```suggestion
    s32 ret = false;

    if (func_80AF8ED4(this, globalCtx, arg2, ACTORCAT_NPC, ACTOR_EN_AN)) {
        func_8013AED4(&this->unk_356, 3, 7);
        this->unk_356 |= 0x20;
        this->unk_356 |= 0x9000;
        if (this->unk_258 != 0) {
            this->unk_356 |= 0x800;
            func_80AF7E98(this, 5);
        } else {
            func_80AF7E98(this, 3);
        }
        ret = true;
```

**Review by engineer124 — COMMENTED** [review]
Great work with this! Only some very minor comments

**Review by engineer124 — APPROVED** [review]

File: src/overlays/actors/ovl_En_Pm/z_en_pm.h [hunk 11]
```diff
@@ -6,12 +6,56 @@
...
+    /* 0x0372 */ s16 unk_372;
+    /* 0x0374 */ s16 unk_374;
+    /* 0x0376 */ s16 unk_376;
+    /* 0x0378 */ s16 unk_378;
+    /* 0x037C */ EnPmFunc unk_37C;
+    /* 0x0380 */ s32 unk_380;
+    /* 0x0384 */ s32 unk_384;
+    /* 0x0388 */ s32 unk_388;
+    /* 0x038C */ s32 unk_38C;
+    /* 0x0390 */ char unk390[0x4];
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_Pm/z_en_pm.h:55]:
```suggestion
    /* 0x0390 */ UNK_TYPE1 unk390[0x4];
```

File: src/overlays/actors/ovl_En_Pm/z_en_pm.h [hunk 12]
```diff
@@ -6,12 +6,56 @@
...
+    /* 0x0394 */ s32 unk_394;
+    /* 0x0398 */ s32 unk_398;
+    /* 0x039C */ char unk39C[0x4];
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_Pm/z_en_pm.h:58]:
```suggestion
    /* 0x039C */ UNK_TYPE1 unk39C[0x4];
```

**Review by Kenix3 — CHANGES_REQUESTED** [review]

**Review by AngheloAlf — APPROVED** [review]
Looks good. Don't forget to address the other reviews and to fix conflicts

File: src/overlays/actors/ovl_En_Suttari/z_en_suttari.c [hunk 13]
```diff
@@ -748,20 +748,19 @@ s32 func_80BABF64(EnSuttari* this, GlobalContext* globalCtx, struct_80133038_arg
 }
 
 s32 func_80BABFD4(EnSuttari* this, GlobalContext* globalCtx) {
-    UNK_TYPE1 sp7C[0x424];
+    s32 pad3[9];
+    f32 sp7C[256];
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Suttari/z_en_suttari.c:752]:
```suggestion
    f32 sp7C[265];
```
Should be the same as En_Pm

**Review by hensldm — CHANGES_REQUESTED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**